File: src/main/java/com/github/bjoernpetersen/jmusicbot/InitializationException.java
package com.github.bjoernpetersen.jmusicbot;

/**
 * An exception thrown by Plugin initialization methods.
 */
public final class InitializationException extends Exception {

  public InitializationException() {
    super();
  }

  public InitializationException(String message) {
    super(message);
  }

  public InitializationException(String message, Throwable cause) {
    super(message, cause);
  }

  public InitializationException(Throwable cause) {
    super(cause);
  }
}

File: src/main/java/com/github/bjoernpetersen/jmusicbot/InitializationException.kt
package com.github.bjoernpetersen.jmusicbot

import java.lang.Exception

/**
 * An Exception occurring during the initialization of the MusicBot / a Plugin.
 */
class InitializationException : Exception {

  constructor() : super()
  constructor(message: String) : super(message)
  constructor(message: String, cause: Throwable) : super(message, cause)
  constructor(cause: Throwable) : super(cause)
}

/**
 * An Exception thrown if the user cancels the MusicBot initialization.
 */
class CancelException : Exception {

  constructor() : super()
  constructor(message: String) : super(message)
  constructor(message: String, cause: Throwable) : super(message, cause)
  constructor(cause: Throwable) : super(cause)
}
